owsi-core/owsi-core-components/owsi-core-component-jpa/src/main/java/fr/openwide/core/jpa/batch/executor/AbstractBatch.java

```diff
@@ -0,0 +1,48 @@
+package fr.openwide.core.jpa.batch.executor;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.interceptor.DefaultTransactionAttribute;
+import org.springframework.transaction.interceptor.TransactionAttribute;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import fr.openwide.core.jpa.business.generic.service.IEntityService;
+import fr.openwide.core.jpa.search.service.IHibernateSearchService;
+import fr.openwide.core.jpa.util.EntityManagerUtils;
+
+public abstract class AbstractBatch<T extends AbstractBatch<T>> {
```

Comment:
I will sound pedantic again, I guess, but the term "BatchExecutor" ("AbstractBatchExecutor", in this case) seemed to fit better to me...
This class is about orchestrating some treatment that takes a batch of objects as an input. So it's not a batch itself, it just deals with a batch.
Plus, the term "BatchExecutor" makes it clear that the purpose of this class is quite similar to that of java.util.concurrent.Executor, only with some specific aspects related to batches.
Of course, same goes for child classes.

Would it pose a problem if I renamed this class (and its subclasses)?